api/src/main/java/org/pmiops/workbench/exfiltration/EgressRemediationService.java

```diff
@@ -155,6 +176,23 @@ private int getEgressIncidentCountForUser(DbUser user) {
     return incidentCount;
   }
 
+  private boolean shouldNotifyForEvent(DbEgressEvent event) {
+    if (event.getWorkspace() == null) {
+      return true;
+    }
+
+    Timestamp debounceAfter =
+        Timestamp.from(event.getCreationTime().toInstant().minus(EGRESS_NOTIFY_DEBOUNCE_TIME));
+    List<DbEgressEvent> priorEvents =
+        egressEventDao.findAllByUserAndWorkspaceAndCreationTimeBetweenAndEgressEventIdNot(
+            event.getUser(),
+            event.getWorkspace(),
+            debounceAfter,
+            event.getCreationTime(),
+            event.getEgressEventId());
+    return priorEvents.isEmpty();
+  }
```

Comment:
I decided to add some debouncing behavior, I'm worried about spamming the user with emails when multiple windows are tripped.

PTAL